internal/provider/data_source_hvn_route.go

```diff
@@ -0,0 +1,104 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceHVNRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN Route data source provides information about an existing HVN route.",
+		ReadContext: dataSourceHVNRouteRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &hvnRouteDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"destination_cidr": {
+				Description: "The destination CIDR of the HVN route",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			// Computed outputs
+			"self_link": {
+				Description: "A unique URL identifying the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"target": {
+				Description: "The target of the HVN route.",
+				Type:        schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed: true,
+			},
+			"state": {
+				Description: "The state of the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time that the HVN route was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceHVNRouteRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	hvn := d.Get("hvn").(string)
+	var hvnLink *sharedmodels.HashicorpCloudLocationLink
+
+	hvnLink, err := parseLinkURL(hvn, "hashicorp.network.hvn")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+	destination := d.Get("destination_cidr").(string)
+
+	log.Printf("[INFO] Reading HVN route for HVN (%s) with destination_cidr=%s ", hvn, destination)
+	route, err := clients.ListHVNRoutes(ctx, client, hvnLink.ID, destination, "", "", loc)
+	if err != nil {
+		return diag.Errorf("unable to retrieve HVN route for HVN (%s) with destination_cidr=%s: %v",
+			hvn, destination, err)
+	}
+
+	// ListHVNRoutes call should return 1 and only 1 HVN route.
+	if len(route) > 1 {
+		return diag.Errorf("Unexpected number of HVN routes returned for destination_cidr=%s: %d", destination, len(route))
```

Comment:
Can you help me understand this error? cloud-network removed it's overlapping CIDR validation, right? Could a customer create two HVN Routes with the same CIDR?

Reply:
Customers can't create HVN routes with identical CIDRs, however in the case this changes in the future (however unlikely), we should still handle the case of ListHVNRoutes returning more than 1 route

Reply:
we won't have routes with identical CIDRs, there's no way to know how to route traffic in this case

Reply:
@smaant linked me to [this DB trigger](https://github.com/hashicorp/cloud-network/blob/b39b5a6d46b4caf1a42a6132e821cf66d1b2b62a/models/migrations/20210323110208_routes_uniqueness_check.up.sql). I was missing what allowed overlapping, but enforced unique CIDRs.